### ISSUE TITLE
Elgg3: Using literal MySQL statements in 'wheres' options parameter is deprecated

### ISSUE THREAD

**Issue by rohit1290** [Elgg/Elgg#12400]
```
ELGG.WARNING: Deprecated in 3.0: Using literal MySQL statements in 'wheres' options parameter is deprecated. Instead use a closure that receives an instanceof of QueryBuilder and returns a composite DBAL expression {{ e.guid != 2418 }} 
Called from [#32] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/LegacyQueryOptionsAdapter.php:953\n 
-> [#31] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/LegacyQueryOptionsAdapter.php:70  {"backtrace":["
[#32] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/LegacyQueryOptionsAdapter.php:953","
[#31] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/LegacyQueryOptionsAdapter.php:70","
[#30] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/QueryOptions.php:80","
[#29] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/Repository.php:28","
[#28] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/Repository.php:73","
[#27] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/Repository.php:87","
[#26] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/lib/entities.php:546","
[#25] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/ElggBatch.php:306","
[#24] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/ElggBatch.php:380","
[#23] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/ElggEntity.php:1654","
[#22] unknown","
[#21] /opt/lampp/htdocs/www/elgg_30rc2/vendor/php-di/invoker/src/Invoker.php:82","
[#20] /opt/lampp/htdocs/www/elgg_30rc2/vendor/php-di/php-di/src/DI/Container.php:256","
[#19] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Invoker.php:71","
[#18] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/lib/elgglib.php:1177","
[#17] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/ElggEntity.php:1665","
[#16] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Database/Plugins.php:307","
[#15] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/lib/plugins.php:18","
[#14] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/views/default/admin/plugins.php:15","
[#13] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php:522","
[#12] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/ViewsService.php:455","
[#11] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/lib/views.php:251","
[#10] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/lib/admin.php:645","
[#9] unknown","
[#8] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Router.php:291","
[#7] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Router.php:251","
[#6] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Router.php:124","
[#5] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Router.php:98","
[#4] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Application.php:466","
[#3] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Application.php:362","
[#2] /opt/lampp/htdocs/www/elgg_30rc2/vendor/elgg/elgg/engine/classes/Elgg/Application.php:418","
[#1] /opt/lampp/htdocs/www/elgg_30rc2/index.php:8"],"process_id":7493,"memory_peak_usage":"2 MB","memory_usage":"2 MB","url":"/www/elgg_30rc2/admin/plugins","ip":"::1","http_method":"GET","server":"localhost","referrer":"http://localhost/www/elgg_30rc2/admin/plugins","unique_id":"XFWS5x6Nli1A-d4pat8cFQAAAAs"}\n, 
referer: http://localhost/www/elgg_30rc2/admin/plugins
```

**Comment by jdalsem** [comment]
https://github.com/Elgg/Elgg/blob/3.0/engine/classes/ElggEntity.php#L1642

**Comment by jeabakker** [comment]
fixed in #12455
